lib/bright/chats.ex

```diff
@@ -67,6 +67,9 @@ defmodule Bright.Chats do
       end)
 
     list_chats_interview(user_id, status)
+    |> Enum.concat(list_chats_coordination(user_id, [:waiting_recruit_decision]))
+    |> Enum.concat(list_chats_employment(user_id, [:waiting_response]))
+    |> Enum.sort_by(& &1.updated_at, :desc)
```

Comment:
更新時間でソート

Reply:
通りすがりですが、structなデータの並び替えは、`compare`実装のあるモジュールを指定する必要があります。たぶん `DateTime`あたりの指定が必要と思います。

https://hexdocs.pm/elixir/1.13.4/Enum.html#sort_by/3

> As in [sort/2](https://hexdocs.pm/elixir/1.13.4/Enum.html#sort/2), avoid using the default sorting function to sort structs, as by default it performs structural comparison instead of a semantic one. In such cases, you shall pass a sorting function as third element or any module that implements a compare/2 function. For example, to sort users by their birthday in both ascending and descending order respectively:

Reply:
@tato-wk 
検証してみましたが、結果が同じでした（期待の結果だった）
問題になる具体的なケースありますか？

Reply:
@ymn-t-yamanashi 

https://hexdocs.pm/elixir/1.17.2/Enum.html#sort/2-sorting-structs
の例を取ると下記のような感じです。

```
    test "dummy" do
      chat_1 = insert(:recruit_chat, updated_at: ~N[2019-01-01 00:00:00])
      chat_2 = insert(:recruit_chat, updated_at: ~N[2020-03-02 00:00:00])
      chat_3 = insert(:recruit_chat, updated_at: ~N[2019-06-06 00:00:00])

      expected = [
        ~N[2020-03-02 00:00:00],
        ~N[2019-06-06 00:00:00],
        ~N[2019-01-01 00:00:00]
      ]

      # bad
      actual_bad =
        Enum.sort_by([chat_1, chat_2, chat_3], & &1.updated_at, :desc)
        |> Enum.map(& &1.updated_at)

      # good
      actual_good =
        Enum.sort_by([chat_1, chat_2, chat_3], & &1.updated_at, {:desc, NaiveDateTime})
        |> Enum.map(& &1.updated_at)

      refute expected == actual_bad
      assert expected == actual_good
    end
```

Reply:
@tato-wk 
なるほど、日付から比較してしまうのね
たまたま、月をを超えたデータが無いから再現しなかっただけですね
ありがとうございました